### PR TITLE
feat(massive_act): add module-position activation and residual-write diagnostics

### PR DESCRIPTION
**Pull request by MrCapricornLiu**
## Summary

- Record activation statistics at five executed module positions:
  - layer input
  - attention output
  - post-attention residual
  - FFN/MoE output
  - post-FFN residual
- Report RMS, absolute maximum, absolute p99, and outlier ratio at each position.
- Add attention and FFN residual-update RMS ratios based on actual forward boundaries.
- Document the new metrics and their definitions.

## Motivation

Layer-level activation statistics alone cannot determine whether an outlier originates from Attention, FFN/MoE, or residual accumulation. The new positions provide module-level localization and measure the branch contribution actually written into the residual stream.

The implementation observes executed boundaries instead of reconstructing outputs manually, covering bias, dropout, BDA, and hyper-connection residual mixing.

## Validation

- Relevant PaddleFleet monitor suite: 120 passed.
- Covered standard residual and mHC-compatible execution boundaries, metric values, and hook cleanup.

### CODE REVIEW

File: src/internal_medicine/backends/paddlefleet/massive_activation_monitor.py [hunk 1]
```diff
@@ -190,13 +240,95 @@ def hook_fn(module, inputs):
...
+        metric_name: str,
+        before: paddle.Tensor,
+        after: paddle.Tensor,
+        attn_type: str | None,
+    ) -> None:
+        if tuple(before.shape) != tuple(after.shape):
+            return
+        before_fp32 = before.detach().astype("float32")
+        update_fp32 = after.detach().astype("float32") - before_fp32
+        ratio = paddle.sqrt(update_fp32.square().mean()) / paddle.sqrt(before_fp32.square().mean()).clip(min=1e-30)
```
Comment (yuweivvv) [inline src/internal_medicine/backends/paddlefleet/massive_activation_monitor.py:274]: 缺乏try/except异常保护，出现问题时会中断模型训练。

**Review by yuweivvv — CHANGES_REQUESTED** [review]

File: src/internal_medicine/backends/paddlefleet/massive_activation_monitor.py (hunk 1 above)
Comment (yuweivvv) [reply src/internal_medicine/backends/paddlefleet/massive_activation_monitor.py:274]: 包括但不限于这一块，请全面排查修复同类问题。

Comment (MrCapricornLiu) [reply src/internal_medicine/backends/paddlefleet/massive_activation_monitor.py:274]: Thanks for the feedback. I have comprehensively audited all hooks in this PR, added fail-closed exception protection for the newly introduced hook paths, rebased onto the latest master, and verified the changes with full regression tests and a real PaddleFleet GPU training canary. The PR is ready for re-review.
